src/models/adapters.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "../generators.h"
+#include "model.h"
+
+namespace Generators {
+
+Adapter::Adapter(const char* adapter_file_path, Ort::Allocator* allocator)
+    : adapter_{OrtLoraAdapter::Create(fs::path(adapter_file_path).c_str(), *allocator)} {}
+
+const OrtLoraAdapter* Adapter::AcquireRef() {
```

Comment:
Are the AcuireRef() and ReleaseRef() management entirely inside our code? A user can't mess these up?

Reply:
Users don't have access to individual adapters. The adapters that are active have ref count > 0 where as the adapters that are inactive have ref count = 0.

When the user wants to unload an adapter, they must have no generator object that has that same adapter active. Otherwise, the unload will fail.

However, if we inside our code try to AcquireRef and forget to ReleaseRef (i.e. there is mismatch between AcquireRef and ReleaseRef within our code), it can lead to failures when unload is invoked since the adapter considers itself still active.

The generator holds a shared pointer to the adapters, so even if the user attempts to destroy the adapters container (that holds all adapters), the container won't be destroyed unless all generators that reference it are destroyed.

Reply:
But maybe we can automate this further by the use of shared pointers and use the shared_ptr::use_count variable to determine whether unload can be executed? I can explore this further.

Reply:
I will defer this to a future commit, once we have more clarity on whether we should be ref counting to begin with.